apps/mobile/src/components/CRUDOfferFlow/atoms/offerFormStateAtoms.ts

```diff
@@ -1110,16 +1126,31 @@ export const offerFormMolecule = molecule(() => {
         amountBottomLimit > currencies[currency].maxAmount) ??
       false
 
-    return (
-      (noListingTypeOrOfferType ||
-        noOfferDescription ||
-        priceNotFilled ||
-        deliveryMethodNotFilled ||
-        pickupLocationNotFilled ||
-        exceededLimit ||
-        noLocationForInPersonOffer) ??
-      false
-    )
+    if (exceededLimit) return 'exceededLimit'
+
+    return undefined
+  })
+
+  const emitAlertBasedOnCurrentStepIfAnyAtom = atom(null, (get) => {
+    const {t} = get(translationAtom)
+    const reason = get(dontAllowNavigationToNextStepAndReturnReasonAtom)
+
+    if (reason === 'noListingType')
+      Alert.alert(t('offerForm.errorListingTypeNotFilled'))
+    if (reason === 'noOfferType')
+      Alert.alert(t('offerForm.errorOfferTypeNotFilled'))
+    if (reason === 'noOfferDescription')
+      Alert.alert(t('offerForm.errorDescriptionNotFilled'))
+    if (reason === 'noLocationForInPersonOffer')
+      Alert.alert(t('offerForm.errorLocationNotFilled'))
+    if (reason === 'priceNotFilled')
+      Alert.alert(t('offerForm.errorPriceNotFilled'))
+    if (reason === 'deliveryMethodNotFilled')
+      Alert.alert(t('offerForm.errorDeliveryMethodNotFilled'))
+    if (reason === 'pickupLocationNotFilled')
+      Alert.alert(t('offerForm.errorPickupLocationNotFilled'))
+    if (reason === 'exceededLimit')
+      Alert.alert(t('offerForm.errorExceededLimits'))
```

Comment:
Theoretically, you can return the offer form string right away from the `dontAllowNavigationToNextStepAndReturnReasonAtom` atom. So you don't have to do this, but just 

```
Alert.alert(`offerFor.${reason}`)
```

It just needs to be returned `as const` to make the typechecking happy.